src/app/tests/TestClosureDimensionCluster.cpp

```diff
@@ -0,0 +1,1871 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/clusters/closure-dimension-server/closure-dimension-cluster-logic.h>
+#include <app/clusters/closure-dimension-server/closure-dimension-cluster-objects.h>
+#include <app/clusters/closure-dimension-server/closure-dimension-delegate.h>
+#include <app/clusters/closure-dimension-server/closure-dimension-matter-context.h>
+#include <lib/support/CHIPMem.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <pw_unit_test/framework.h>
+#include <system/SystemClock.h>
+#include <system/SystemTimer.h>
+
+using namespace chip::app::Clusters::ClosureDimension;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ClosureDimension {
+
+namespace {
+
+constexpr uint16_t kExpectedClusterRevision = 1u;
+
+} // namespace
+
+class TestDelegate : public DelegateBase
+{
+public:
+    TestDelegate() {}
+
+    Protocols::InteractionModel::Status HandleSetTarget(const Optional<Percent100ths> & position, const Optional<bool> & latch,
+                                                        const Optional<Globals::ThreeLevelAutoEnum> & speed)
+    {
+        return Protocols::InteractionModel::Status::Success;
+    }
+
+    Protocols::InteractionModel::Status HandleStep(const StepDirectionEnum & direction, const uint16_t & numberOfSteps,
+                                                   const Optional<Globals::ThreeLevelAutoEnum> & speed)
+    {
+        return Protocols::InteractionModel::Status::Success;
+    }
+
+    bool IsManualLatchingNeeded() { return isLatchManual; }
+
+    void ToggleManualLatching(const bool manualLatch) { isLatchManual = manualLatch; }
```

Comment:
Function doesn't implement a toggle.
```suggestion
    void SetManualLatching(const bool manualLatch) { isLatchManual = manualLatch; }
```